core/server/common/src/main/java/alluxio/cli/validation/UserLimitValidationTask.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+ * Task for validating system limit for current user.
+ */
+public final class UserLimitValidationTask implements ValidationTask {
+  private static final int NUMBER_OF_OPEN_FILES_MIN = 4096;
```

Comment:
On [this document page](http://www.alluxio.org/docs/master/en/Building-Alluxio-Master-Branch.html), we suggest user setting `maxproc` to 2048 and `maxfiles` to 16384. Should we use that as the lower bounds? Using bigger numbers will make this validateEnv command fail when user followed the document recommendation. Or we can also update the document to recommend bigger numbers.